packages/video_player/video_player_avfoundation/darwin/video_player_avfoundation/Sources/video_player_avfoundation/FVPVideoPlayerPlugin.m

```diff
@@ -345,13 +345,7 @@ - (void)observeValueForKeyPath:(NSString *)path
     AVPlayerItem *item = (AVPlayerItem *)object;
     switch (item.status) {
       case AVPlayerItemStatusFailed:
-        if (_eventSink != nil) {
-          _eventSink([FlutterError
-              errorWithCode:@"VideoError"
-                    message:[@"Failed to load video: "
-                                stringByAppendingString:[item.error localizedDescription]]
-                    details:nil]);
-        }
+        [self sendFailedToLoadVideoEvent:item.error];
```

Comment:
nit: `sendFailedToLoadVideoEventIfNeeded` since you moved the check inside

Reply:
I would not say "if needed" sounds right here. Meaning of that check is rather like "if can" or "if possible".